stan/math/prim/mat/fun/matrix_exp_multiply.hpp

```diff
@@ -18,11 +18,12 @@ namespace math {
 template <int Cb>
 inline Eigen::Matrix<double, -1, Cb> matrix_exp_multiply(
     const Eigen::MatrixXd& A, const Eigen::Matrix<double, -1, Cb>& B) {
-  check_multiplicable("scale_matrix_exp_multiply", "A", A, "B", B);
-  check_square("scale_matrix_exp_multiply", "input matrix", A);
-  if (A.size() == 0)
+  if (A.size() == 0 && B.size() == 0)
```

Comment:
Should this be 
```cpp
if (A.size() == 0 || B.size() == 0)
```

Reply:
If `A.size() == 0` but `B.size() != 0` then `exp(A)` and `B` are not multiplicable, and we should throw instead of returning an empty matrix. Same thing if If `A.size() != 0` and `B.size() == 0`. The only case where the multiplication can be done if A or B are 0 size is when the other is 0 size as well.

Reply:
Ah yeah, forgot about that A must be square. How about the case where A(5,5) and B(5,0)? Not saying this is wrong, just a question.

Reply:
Or does the matrix_exp_action_handler handle that case as expected?

Reply:
This case would cause `check_multiplicable` to throw because B has 0 columns. This is perhaps overly strict, but I'm not sure if there is any value in allowing it, as it would have failed even if we had A * B rather than exp(A) * B.

Reply:
Ah ok, I actually was not aware that check_multiplicable checks that rows of A > 0 and cols of B > 0. So this one is on me. 

I don't think that this behavior is the best as its not clear from the name that it does this, but that is a separate issue, definitely out of scope for this PR.

Sorry for the confusion here. This is good to go. Thanks again!